hack/lib/build/binaries.sh

```diff
@@ -182,9 +182,15 @@ os::build::internal::build_binaries() {
     # Use eval to preserve embedded quoted strings.
     eval "goflags=(${OS_GOFLAGS:-})"
 
+    local brand=""
     local arg
+
     for arg; do
-      if [[ "${arg}" == -* ]]; then
+      if [[ "${arg}" == "-ocp" ]]
```

Comment:
Instead of using GOFLAGS and checking for a specific value, it would be better to define an entirely different env var for tags. Then we don't need the magic check here.

We should confirm that there isn't a way to set this already (OS_GOFLAGS_TAGS?)

Reply:
I did try a version where I had a new GOFLAG_TAGS.  I still had do checking on it and I had to add new code to the Makefile to handle another env flag var. It turned out to be easier to just use the existing GOFLAG and check it for the -ocp value.  I am open to discussion, but I thought it isolated changes to just the binaries.sh script.  
As for the OS_GOFLAG_TAGS (which is set automatically), if you mean forcing the brand based on os/arch I am worried that someone might want to have okd on RHEL. Or did you have something else in mind @spadgett ?

Reply:
That is the solution I tried at first, but the problem is the values in in the ADDITIONAL_GOTAGS get processed by the same code as goflags, and if there is no "-", the value is  considered a package to get built. So `make WHAT="cmd/console" GOFLAGS="" ADDITIONAL_GOTAGS="ocp"` gives an error : can't load package: package github.com/openshift/console-operator/ocp. I need to have the code above ignore items from the ADDITIONAL_GOTAGS but I was not sure of an easy way to do that.

Reply:
Ok I see the problem. Yeah, my suggestion is wrong. Let’s catch up on it tomorrow

Reply:
Does this work in the Dockerfile (without the Makefile changes)?

```
RUN ADDITIONAL_GOTAGS="-ocp" make build
```

Reply:
No, without the hack/build-go.sh $(WHAT) $(GOFLAGS) $(ADDITIONAL_GOTAGS) in the Makefile, that flag is ignored

Reply:
Also, does anyone know if its common to update scripts in `hack/lib` or at that point is the script typically copied out of `hack/lib` into `hack/` (top level) and tweaked as non-lib?  I'm guessing there isn't a best practice here (though all repos more or less use this.... and `lib` implies it should be generic).

Reply:
Yeah, I did not want to update those files (hack/lib/build/binaries.sh) but there was no obvious mechanism to get additional flags into the tag field.

Reply:
@zherman0 It works for me. I did export ADDITIONAL_GOTAGS from the Makefile

```
~/go/src/github.com/openshift/console-operator goflags*
❯ ADDITIONAL_GOTAGS=ocp make
hack/build-go.sh
++ Building go targets for darwin/amd64: cmd/console
[INFO] hack/build-go.sh exited with code 0 after 00h 00m 05s

~/go/src/github.com/openshift/console-operator goflags*
❯ ./_output/local/bin/darwin/amd64/console version
ConsoleOperator v0.0.1
Git Commit: 275c5a58
Build Date: 2019-02-06T20:40:28Z
Current Brand Setting: ocp
```

Reply:
I opened a PR with the change: https://github.com/zherman0/console-operator/pull/1

Reply:
To answer on the original `hack/` scripts question, there does not seem to be a goal to keep them consistent across repos.  There are hard-coded bits that have to be changed for each repo.